src/robot/api/deco.py

```diff
@@ -0,0 +1,40 @@
+# -*- coding: utf-8 -*-
+
+#  Copyright 2008-2014 Nokia Solutions and Networks
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+
+def robot_name(custom_name=None):
+    """
+    A decorator for keywords which allows providing a custom keyword name that Robot
+    should use. For example:
+    myuserlibrary.py:
+    ...
+    class MyUserLibrary:
+        ...
+        @robot_name('Login Via User Panel')
+        def login(username, password):
+           ...
+
+    mytest.robot:
+    ...
+    Login Via User Panel  myusername  mypassword
+    ...
+
+    :param custom_name: a custom name for the decorated keyword
+    """
```

Comment:
1) Would be good to start the docstring with a summary line as explained at https://www.python.org/dev/peps/pep-0257/#multi-line-docstrings.

2) The doc should also mention that the decorator adds `robot_name` attribute. 

3) For [Sphinx to format the example nicely](http://sphinx-doc.org/markup/code.html), it should be indented and there should be `::` before it.

4) Run `doc/api/generate.py` to generate API docs and see how documentation looks like. Notice that you get plenty of warnings that can be ignored when you generate docs.